pycbc/psd/variation.py

```diff
@@ -210,10 +210,197 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
+        an increments number of seconds prior to the end of the input strain
+        data.
+
+    """
+    sample_rate = int(strain.sample_rate)
+
+    # Grab the last increments worth of data with extra to account for the need
+    #  to trim the data to remove edge effects.
+    astrain = strain.time_slice(strain.end_time - increment - (data_trim*3),
+                                strain.end_time)
+
+    # Convole the data and the filter to produce the PSD variation timeseries,
+    #  then trim the beginning and end of the data to prevent edge effects.
+    wstrain = sig.fftconvolve(astrain, full_filt, mode='same')
+    wstrain = wstrain[int(data_trim * sample_rate):-int(data_trim * sample_rate)]
+
+    # Create a PSD variation array by taking the mean square of the PSD
+    #  variation timeseries every short_stride
+    short_ms = numpy.mean(
+        wstrain.reshape(-1, int(sample_rate * short_stride)) ** 2, axis=1)
+
+    # Create an array of averages to substitute out outliers in the PSD
+    #  variation array
+    ave = 0.5 * (short_ms[2:] + short_ms[:-2])
+    outliers = short_ms[1:-1] > (2. * ave)
+    short_ms[1:-1][outliers] = ave[outliers]
+
+    # Calculate the average of the PSD variation array for every second
+    m_s = []
+    stride = 1 / short_stride
+    for idx in range(int(len(short_ms) / stride)):
+        start = int(stride * idx)
+        end = int(stride * (idx + 1))
+        m_s.append(numpy.mean(short_ms[start:end]))
+
+
+    # Convert m_s to a numpy array
+    m_s = numpy.array(m_s, dtype=wstrain.dtype)
+
+    # Convert the m_s numpy array to a pycbc timeseries which now contains the
+    #  psd variation value every second.
+    psd_var = TimeSeries(m_s,
+                         delta_t=1.0,
+                         epoch=strain.end_time - increment - (data_trim * 2))
+
+    return psd_var
+
+
+def live_find_var_value(triggers,
+                        psd_var_timeseries):
+    """
+    Interpolate between PSD variation values to find the PSD variation value
+    associated with a specific trigger.
+
+    Parameters
+    ----------
+    triggers : dict
+        A dictionary containing the trigger values to find the PSD variation
+        of.
+    psd_var_timeseries : pycbc.timeseries
+        A timeseries containing the PSD variation value for each second of the
+        latest increment in PyCBC Live. Created by live_calc_psd_variation.
+
+    Returns
+    -------
+    psd_var_vals : numpy.ndarray
+        A numpy array containing the PSD variation values associated with the
+        triggers.
+    """
+
+    # Find gps time of the trigger
+    trigger_times = triggers['end_time']
+
+    # Interpolate between values
+    interpolator = interp1d(psd_var_timeseries.sample_times.numpy(),
+                            psd_var_timeseries.numpy(),
+                            fill_value=1.0,
+                            bounds_error=False)
+    psd_var_vals = interpolator(trigger_times)
```

Comment:
Actually you don't need to create the variable 'trigger_times' as you only use it once. Can just do this directly IMO. (Then as the line is slightly more complex you can put a comment!)
```suggestion
    # Evaluate at the trigger times
    psd_var_vals = interpolator(triggers['end_time'])
```